### PR TITLE
Feat/merchant csv export

### PR DESCRIPTION
**Pull request by BanerjeeAgniva**
# Pull Request

> **Note on scope:** despite the branch name, this PR is **not** the CSV export — that already merged in #147 and is on `main`. Because the stacked PRs were merged into their parent branches, `feat/merchant-csv-export` picked up the pickup-slot-cap commits (via #148), so **this PR's actual diff is the per-slot pickup capacity cap (feature #11)** landing on `main`.

## Summary
Add a **per-slot pickup capacity cap**: a booking whose chosen pickup slot is already at capacity is rejected **before payment**, so a full slot never charges the customer. Enforced identically in the B2C and B2B booking flows.

---

## What Changed
- New `PickupSlotCapacity` component, called pre-payment in both `BookingServiceImpl` (B2C) and `B2bBookingServiceImpl` (B2B).
- Cap is an ops-tunable property `orders.pickup-slot.max-per-slot` (default 50) — `PickupSlotProperties`.
- A full or invalid slot throws `PickupSlotFullException` → **HTTP 409** (wired in `OrdersGlobalExceptionHandler`).
- Only **`DA_PICKUP`** shipments count against a slot — a `SELF_DROP` parcel needs no DA, so it neither consumes nor is capped (new filtered repo query `countByOriginCityAndScheduledPickupStartAndPickupTypeAndCancelledAtIsNull`).
- Unit tests over the ASAP / has-room / full / invalid-hour / self-drop-skip branches.

---

## Why This Change?
On a one-day SLA, a pickup time-window has finite DA capacity. Without a cap, unlimited bookings can pile into a single (city, slot) window that can't physically be served. Doing the check **before** payment means a full slot rejects cleanly (409) instead of charging the customer and failing later.

`ponytail:` the cap is a **soft** cap (count-then-book) — concurrent writers can overbook a slot by up to (concurrent bookings − 1). The hard-cap upgrade path (per-slot counter row + row lock, mirroring `VanManifestServiceImpl`) is noted in-code; add it only if overbooking bites at pilot volume.

---

## Testing
- [x] Tested locally (`mvn test -pl orders`)
- [x] Edge cases considered (ASAP, self-drop, invalid slot hour; concurrency documented as a soft cap)
- [x] No breaking changes introduced

### Test Evidence
```
mvn -pl orders -am test -Dtest=PickupSlotCapacityTest,BookingServiceImplTest,B2bBookingServiceImplTest
...
Tests run: 5,  Failures: 0, Errors: 0, Skipped: 0 -- PickupSlotCapacityTest
Tests run: 14, Failures: 0, Errors: 0, Skipped: 0 -- BookingServiceImplTest
Tests run: 9,  Failures: 0, Errors: 0, Skipped: 0 -- B2bBookingServiceImplTest
BUILD SUCCESS
```
CodeRabbit/claude review on the original #148 (the SELF_DROP-vs-DA_PICKUP gap) was addressed by the `countBy…PickupType…` filter included here.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: orders/src/main/java/com/oneday/orders/service/impl/PickupSlotCapacity.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            return;   // self-drop consumes no DA pickup slot
+        }
+        String city = originCity.toUpperCase();
+        Instant slotStart = PickupSlots.resolve(slotDate, startHour).start();
+        int booked = shipmentRepository
+                .countByOriginCityAndScheduledPickupStartAndPickupTypeAndCancelledAtIsNull(
+                        city, slotStart, PickupType.DA_PICKUP);
+        if (booked >= props.getMaxPerSlot()) {
+            throw new PickupSlotFullException(
+                    "The " + startHour + ":00 pickup slot for " + city + " is full. Please choose another slot.");
```
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/java/com/oneday/orders/service/impl/PickupSlotCapacity.java:60]: Deliberate soft cap for v1 — flagged in code with a `ponytail:` comment naming the ceiling and the upgrade path. The count-then-book window can overbook a slot by up to (concurrent bookings − 1). A pickup-slot cap is an operational capacity hint, not a hard invariant like payment, and mild overbooking under rare concurrency self-corrects operationally, so we chose not to pay for a per-slot lock/counter at pilot volume. The upgrade path (per-slot reservation row + row lock, mirroring `VanManifestServiceImpl.bindEarliest`) is documented in-code and can be swapped in if overbooking actually bites. Leaving open as a tracked, accepted tradeoff.

File: orders/src/main/java/com/oneday/orders/service/impl/BookingServiceImpl.java [hunk 2]
```diff
@@ -194,6 +197,10 @@ public BookingResponse book(BookingRequest req, String idempotencyKey, String us
         int chargeableWeightGrams = priced.chargeableWeightGrams();
         QuoteResult quote = priced.quote();
 
+        // ── 3b. Pickup-slot capacity (before payment, so a full slot never charges) ──
+        pickupSlotCapacity.ensureRoom(req.getOriginCity(), req.getPickupSlotDate(),
+                req.getPickupSlotStartHour(), req.getPickupType());
```
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/java/com/oneday/orders/service/impl/BookingServiceImpl.java:202]: Valid observation. The cart/bulk checkout path (`CartServiceImpl` → `bookSettled`) is intentionally not slot-capped in this slice — only the single B2C/B2B `book()` paths are. Unlike single bookings, cart checkout captures one aggregate payment upfront for N items, so a per-item pre-payment check raises an unsettled product question (reject the whole cart if one item’s slot is full, or partially book?). Rather than bake that policy in here, single-booking capping ships now and cart-flow capping is a tracked follow-up — not silently dropped.

[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/java/com/oneday/orders/service/impl/BookingServiceImpl.java:202]: open github issue
